### PR TITLE
Use gRPC Security Token Service (STS) to get call credentials

### PR DESCRIPTION
**Pull request by JimmyCYJ**
Add a new field ```sts_service``` into GoogleGrpc call credential options which support Envoy to exchange token. See https://github.com/grpc/grpc/pull/19032 and https://github.com/grpc/grpc/pull/19587.

Signed-off-by: JimmyCYJ jimmychen.0102@gmail.com

Risk Level: Low
Testing: Unit test

### CODE REVIEW

File: api/envoy/api/v2/core/grpc_service.proto [hunk 1]
```diff
@@ -86,6 +86,27 @@ message GrpcService {
         }
       }
 
+      // [#next-free-field: 10]
+      message StsService {
+        string token_exchange_service_uri = 1 [(validate.rules).string = {uri: true}];
```
Comment (htuch) [inline api/envoy/api/v2/core/grpc_service.proto]: Can you add comments to all these fields?

File: api/envoy/api/v2/core/grpc_service.proto [hunk 2]
```diff
@@ -86,6 +86,27 @@ message GrpcService {
         }
       }
 
+      // [#next-free-field: 10]
+      message StsService {
```
Comment (htuch) [inline api/envoy/api/v2/core/grpc_service.proto]: What's the situation with documentation?

**Comment by JimmyCYJ** [comment]
not stale

File: api/envoy/api/v2/core/grpc_service.proto (hunk 2 above)
Comment (JimmyCYJ) [reply api/envoy/api/v2/core/grpc_service.proto]: The comment ```// [#next-free-field: 10]``` was added by command ```./tools/proto_format.sh fix```, I removed it and added ```--no-verify``` to push the commit.

File: api/envoy/api/v2/core/grpc_service.proto (hunk 1 above)
Comment (JimmyCYJ) [reply api/envoy/api/v2/core/grpc_service.proto]: I have added comments to all these fields. Thanks.

File: api/envoy/api/v2/core/grpc_service.proto (hunk 2 above)
Comment (lizan) [reply api/envoy/api/v2/core/grpc_service.proto]: @JimmyCYJ next-free-id is required, also you should add note how this message is used.

Comment (JimmyCYJ) [reply api/envoy/api/v2/core/grpc_service.proto]: @lizan I have updated comments to the message. Please take a look. Thanks!

File: api/envoy/api/v2/core/grpc_service.proto [hunk 3]
```diff
@@ -86,6 +86,46 @@ message GrpcService {
         }
       }
 
+      // Security token service configuration that allows Google gRPC to
+      // fetch security token from an OAuth 2.0 authorization server.
+      // See https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16 and
+      // https://github.com/grpc/grpc/pull/19587.
```
Comment (lizan) [inline api/envoy/api/v2/core/grpc_service.proto]: Seems this feature is marked experimental in gRPC, how do we want to handle this for API stability? @htuch WDYT? should we create a v2alpha for this one?

Comment (JimmyCYJ) [reply api/envoy/api/v2/core/grpc_service.proto]: @jboeuf is there any timeline to make this a stable feature?

Comment (jboeuf) [reply api/envoy/api/v2/core/grpc_service.proto]: There's no timeline that I'm aware of just yet. I think that we would like first to make sure that the feature is going to work for your use case. Once this is confirmed, we'll probably need a gRPC RFC for it graduate.

@jiangtaoli2016 FYI.

Comment (htuch) [reply api/envoy/api/v2/core/grpc_service.proto]: I think it's fine to use it as is; folks using Google gRPC needs to be aware of the caveats of the external dep.

**Pull request merged**